### PR TITLE
Cs single post

### LINKED ISSUE #155: cross server /posts/<postid>

### PR DESCRIPTION
**Pull request by hiufungk**
resolves #155 

can get cross server post
updated frontend copy url button to copy the correct thing

### CODE REVIEW

**Review by htruong4 — APPROVED** [review]

File: backend/api/view/CreatePostView.py [hunk 1]
```diff
@@ -91,13 +91,65 @@ def get_public_posts(self, request):
...
+                except Exception as e:
+                    return Response(e, status.HTTP_400_BAD_REQUEST)
+        # when server make the request
+        elif server_user_exist:
+            try:
+                authorId = request.META["HTTP_X_REQUEST_USER_ID"]
+            except:
+                return Response("Error: X-Request-User-ID header missing", status.HTTP_400_BAD_REQUEST)
+        else:
+            return Response("Error: Request not from invalid place", status.HTTP_400_BAD_REQUEST)
```
Comment (derrick-w) [inline backend/api/view/CreatePostView.py]:
```suggestion
            return Response("Error: Request not from valid server", status.HTTP_400_BAD_REQUEST)
```

File: backend/api/view/CreatePostView.py [hunk 2]
```diff
@@ -109,6 +161,10 @@ def get_public_post_by_id(self, request, post_id):
         except:
             return Response("Error: Post Does Not Exist", status.HTTP_400_BAD_REQUEST)
 
+
```
Comment (derrick-w) [inline backend/api/view/CreatePostView.py:164]: uhh

File: backend/project_404_T5/settings.py [hunk 3]
```diff
@@ -123,7 +123,7 @@
 # make sure it has a slash at the end
 
 #This URL is meant for developer environment
-# BACKEND_URL = "https://127.0.0.1:8000/"
+# BACKEND_URL = "http://127.0.0.1:8000/"
```
Comment (derrick-w) [inline backend/project_404_T5/settings.py:126]: dunno if needs to be changed

**Review by derrick-w — CHANGES_REQUESTED** [review]

File: backend/project_404_T5/settings.py (hunk 3 above)
Comment (hiufungk) [reply backend/project_404_T5/settings.py:126]: it does

**Review by shchau — APPROVED** [review]
Makes sense to me

**Review by forgeno — APPROVED** [review]

**Review by derrick-w — APPROVED** [review]

**Pull request merged**